src/pudl/analysis/allocate_net_gen.py

```diff
@@ -196,43 +197,157 @@
+        # The monthly version is yuuuuge and we only use the annual data for now.
+        assets += [gen_fuel_by_gen_esc_owner]
+
+    return assets
+
+
+allocate_net_gen_assets = [
+    allocated_net_gen_asset
+    for freq in ["AS", "MS"]
+    for allocated_net_gen_asset in allocate_net_gen_asset_factory(
+        freq=freq,
+        io_manager_key="pudl_sqlite_io_manager",
+    )
+]
+
+
 def allocate_gen_fuel_by_generator_energy_source(
-    pudl_out: "pudl.output.pudltabl.PudlTabl", drop_interim_cols: bool = True
-) -> pd.DataFrame:
+    gf: pd.DataFrame,
+    bf: pd.DataFrame,
+    gen: pd.DataFrame,
+    bga: pd.DataFrame,
+    gens: pd.DataFrame,
+    freq: Literal["AS", "MS"],
+    debug: bool = False,
+):
     """Allocate net gen from gen_fuel table to the generator/energy_source_code level.
 
-    Three main steps here:
-     * grab the three input tables from ``pudl_out`` with only the needed columns
-     * associate ``generation_fuel_eia923`` table data w/ generators
-     * allocate ``generation_fuel_eia923`` table data proportionally
+    There are two main steps here:
 
-     The association process happens via :func:`associate_generator_tables`.
+    * associate ``generation_fuel_eia923`` table data w/ generators
+    * allocate ``generation_fuel_eia923`` table data proportionally
 
-     The allocation process (via `allocate_net_gen_by_gen_esc()`) entails
-     generating a fraction for each record within a ``IDX_PM_ESC`` group. We
-     have two data points for generating this ratio: the net generation in the
-     generation_eia923 table and the capacity from the generators_eia860 table.
-     The end result is a `frac` column which is unique for each
-     generator/prime_mover/fuel record and is used to allocate the associated
-     net generation from the `generation_fuel_eia923` table.
+    The association process happens via :func:`associate_generator_tables`.
 
-    Args:
-        pudl_out (pudl.output.pudltabl.PudlTabl): An object used to create the
-            tables for EIA and FERC Form 1 analysis.
-        drop_interim_cols: True/False flag for dropping interim columns which are used
-            to generate the ``net_generation_mwh`` column (they are mostly the
-            ``frac`` column and  net generataion reported in the original
-            ``generation_eia923`` and ``generation_fuel_eia923`` tables) that are
-            useful for debugging. Default is False, which will drop the columns.
+    The allocation process (via `allocate_net_gen_by_gen_esc()`) entails
+    generating a fraction for each record within a ``IDX_PM_ESC`` group. We
+    have two data points for generating this ratio: the net generation in the
+    generation_eia923 table and the capacity from the generators_eia860 table.
+    The end result is a `frac` column which is unique for each
+    generator/prime_mover/fuel record and is used to allocate the associated
+    net generation from the `generation_fuel_eia923` table.
```

Comment:
Some Arg descriptions would be helpful here.

Reply:
Okay, added the args.